web-bluetooth/notifications.js

```diff
@@ -0,0 +1,47 @@
+var myCharacteristic;
+
+function onStartButtonClick() {
+  var serviceUuid = document.getElementById('service').value;
+  if (serviceUuid.startsWith('0x')) {
+    serviceUuid = parseInt(serviceUuid, 16);
+  }
+
+  var characteristicUuid = document.getElementById('characteristic').value;
+  if (characteristicUuid.startsWith('0x')) {
+    characteristicUuid = parseInt(characteristicUuid, 16);
+  }
+
+  log('Requesting Bluetooth Device...');
+  navigator.bluetooth.requestDevice({filters:[{services:[ serviceUuid ]}]})
+  .then(device => device.connectGATT())
+  .then(server => server.getPrimaryService(serviceUuid))
+  .then(service => service.getCharacteristic(characteristicUuid))
+  .then(characteristic => {
+    myCharacteristic = characteristic;
+    return myCharacteristic.startNotifications().then(() => {
+      log('> Notifications started');
+      myCharacteristic.addEventListener('characteristicvaluechanged', handleNotifications);
+    });
+  })
+  .catch(error => {
+    log('Argh! ' + error);
+  });
+}
+
+function onStopButtonClick() {
+  if (myCharacteristic) {
+    myCharacteristic.stopNotifications().then(() => {
+      log('> Notifications stopped');
+      myCharacteristic.removeEventListener('characteristicvaluechanged', handleNotifications);
+    });
+  }
+}
+
+function handleNotifications(event) {
+  var buffer = event.target.value;
+  var data = new DataView(buffer);
+  for (var i = 0, a = []; i < data.byteLength; i++) {
```

Comment:
Could you include a comment about what the source data will look like? A series of bytes, I know, but maybe there's some additional context? And then a comment about how you're string-ifying it here, but how for a "real" use case you might just process the raw bytes?

Right now, the loop lacks context.